src/http/DataQueryEndpoints.js

```diff
@@ -23,28 +23,35 @@ const onRow = (res, unicastMessage, delimiter, format = 'object', version, metri
 }
 
 const streamData = (res, stream, format, version, metrics) => {
-    let delimiter = ''
-    stream.on('data', (row) => {
-        // first row
-        if (delimiter === '') {
-            onStarted(res)
-        }
-        onRow(res, row, delimiter, format, version, metrics)
-        delimiter = ','
-    })
-    stream.on('end', () => {
-        if (delimiter === '') {
-            onStarted(res)
-        }
-        res.write(']')
-        res.end()
-    })
-    stream.on('error', (err) => {
+    try {
+        let delimiter = ''
```

Comment:
If `stream.on` is failing, it seems like a fundamental problem with the code that _should_ cause a crash instead of being wrapped in a 500 error code and sent back to client? I.e. it is an unexpected error that should not really ever occur unless there is a problem with our code.

Reply:
that's true